grpc-protocol/src/main/java/com/linecorp/armeria/common/grpc/protocol/ArmeriaMessageDeframer.java

```diff
@@ -276,8 +279,15 @@ public void deframe(HttpData data, boolean endOfStream) {
         if (dataLength != 0) {
             final ByteBuf buf = data.byteBuf();
             assert unprocessed != null;
-            unprocessed.add(buf);
-            unprocessedBytes += dataLength;
+            if (decodeBase64) {
+                final ByteBuf decoded = Unpooled.wrappedBuffer(Base64.getDecoder().decode(buf.nioBuffer()));
```

Comment:
I believe this won't work if a message spans across HTTP/2 frames since we'll terminate a base64 decode in the middle. It's where I gave up on this ;)

Reply:
Oops, I thought the encoded message does not span across the frames. Let me check how Akka is doing this. 😉

Reply:
Time to write a stateful decoder? :laughing:

Reply:
I guess so. 😅  Let me think a bit more about how to solve this. 😉

Reply:
@anuraaga Should I also consider the case in which the padding is in the middle of a frame? I think there's no such implementations, but just wanted to make sure. 😉

Reply:
I think it can definitely happen, especially when a decorator decides to split frames.

Reply:
Ah yeah both cases for sure

Reply:
:scream:

Reply:
Fixed. 😉